src/parser/Statement.ts

```diff
@@ -253,23 +253,105 @@ export class AugmentedAssignmentStatement extends Statement {
 export class Block extends Statement {
     constructor(options: {
         statements: Statement[];
-        startingRange?: Range;
     }) {
         super();
         this.statements = options.statements;
-        this.startingRange = options.startingRange;
-        this.range = util.createBoundingRange(
-            this.startingRange,
-            ...(this.statements ?? [])
-        );
     }
 
     public readonly statements: Statement[];
-    public readonly startingRange?: Range;
 
     public readonly kind = AstNodeKind.Block;
 
-    public readonly range: Range | undefined;
+    get range(): Range {
+        if (this.statements.length > 0) {
+            return util.createBoundingRange(...(this.statements));
+        }
+        let lastBitBefore: Position;
+        let firstBitAfter: Position;
+
+        if (isFunctionExpression(this.parent)) {
+            lastBitBefore = util.createBoundingRange(
+                this.parent.tokens.functionType,
+                this.parent.tokens.leftParen,
+                ...this.parent.parameters,
```

Comment:
Is there ever a situation where .parameters is null? If so, maybe do something more like this?

```suggestion
                ...(this.parent.parameters ?? [])
```